File: 지현배/0826/20164.py
N = int(input())
minn = 100
maxx = 0
def sol(num, summ):
    if num < 10:
        global maxx, minn
        if num % 2: summ += 1
        if maxx < summ: maxx = summ
        if minn > summ: minn = summ
    elif num < 100:
        if num % 2: summ += 1
        if num // 10 % 2: summ += 1
        sol(num % 10 + num // 10, summ)
    else:
        n = num
        while n:
            if n % 2: summ += 1
            n //= 10
        s = str(num)
        length = len(s)
        for i in range(1, length - 1):
            for j in range(i + 1, length):
                front = int(s[0:i])
                middle = int(s[i:j])
                back = int(s[j:])
                sol(front + middle + back, summ)
sol(N, 0)
print(minn, maxx)

File: 지현배/0826/20164.js
const fs = require("fs");
let input = fs.readFileSync('/dev/stdin').toString().replace("\n", "") - 0;
let min = 100;
let max = 0;
const sol = (num, sum) => {
  if (num < 10) {
    if (num % 2) sum++;
    if (max < sum) max = sum;
    if (min > sum) min = sum;
    return;
  }
  else if (num < 100) {
    sol(num % 10 + Math.floor(num / 10), sum + (num % 2 ? 1 : 0) + (Math.floor(num / 10) % 2 ? 1 : 0));
    return;
  }
  else {
    // 홀수 몇개?
    let n = num;
    while (n) {
      sum += (n % 2 ? 1 : 0);
      n = Math.floor(n / 10);
    }
    let s = num + "";
    let len = s.length;
    // 분할
    for (let i = 1; i < len - 1; i++) {
      for (let j = i + 1; j < len; j++) {
        // 숫자를 나눈다.
        let front = +s.slice(0, i);
        let middle = +s.slice(i, j);
        let back = +s.slice(j);
        sol(front + middle + back, sum);
      }
    }
  }
}
sol(input, 0);
console.log(min, max);